src/app/handlers/command.handler.ts

```diff
@@ -14,30 +30,40 @@ export class CommandHandler extends Handler {
     super();
   }
 
-  public async execute(message: types.IMessage): Promise<void> {
-    const command = this.build(message.content);
+  public async execute(message: types.IMessage | CommandInteraction): Promise<void> {
```

Comment:
instead of a union type, we should (upstream) populate some CommandPayloadCommon so that it has the same data whether it's from a '!' message or a slash command invocation

Reply:
I was thinking about this, and it feels like that would actually introduce more complexity. The reason being is that typescript unions automatically allow you access common properties. Also this is the only function where there's unioned command types. After this point they're separated and narrowed to their specific types. So another abstracted class would essentially be doing the same de-narrowing this function does but it would only be used in a single place.